### PR TITLE
Prioritizing CDS array and saving genome through GFU

### PR DESCRIPTION
**Pull request by samseaver**

### CODE REVIEW

**Review by JamesJeffryes — APPROVED** [review]

**Review by JamesJeffryes — COMMENTED** [review]
So this will give new functions to the CDSs but some tools only look at gene functions. You might want to consider propagating annotations to the genes

**Comment by samseaver** [comment]
@JamesJeffryes Latest commit is to assign CDS function to gene function

File: lib/kb_plant_rast/kb_plant_rastImpl.py [hunk 1]
```diff
@@ -153,22 +168,32 @@ def annotate_plant_transcripts(self, ctx, input):
         output['hit_ftrs']=len(Hit_Proteins)
         output['hit_fns']=len(Hit_Functions)
 
+        #But, if annotating CDS, need to be able to retrieve parent feature
+        parent_feature_index = dict()
+        if(USE_CDS==1):
+            for i in range(len(plant_genome['data']['features'])):
```
Comment (JamesJeffryes) [inline lib/kb_plant_rast/kb_plant_rastImpl.py]: One line option: `parent_feature_index = dict([(f['id'], i) for i, f in plant_genome['data']['features']])`

File: lib/kb_plant_rast/kb_plant_rastImpl.py [hunk 2]
```diff
@@ -153,22 +168,32 @@ def annotate_plant_transcripts(self, ctx, input):
...
 
-        workspace_id = self.dfu.ws_name_to_id(input['input_ws'])
-        save_result = self.dfu.save_objects({"id": workspace_id,
-                                             "objects": [{'name' : input['output_genome'],
-                                                          'data' : plant_genome['data'],
-                                                          'type' : "KBaseGenomes.Genome",
-                                                          'meta' : plant_genome['info'][10]}]})
+        save_result = self.gfu.save_one_genome({'workspace' : input['input_ws'],
+                                                'name' : input['output_genome'],
+                                                'data' : plant_genome['data']});
```
Comment (JamesJeffryes) [inline lib/kb_plant_rast/kb_plant_rastImpl.py]: Add `'upgrade': 1` to the params to make sure your "function" field is converted to match new genomes

File: lib/kb_plant_rast/kb_plant_rastImpl.py [hunk 3]
```diff
@@ -59,7 +61,20 @@ def annotate_plant_transcripts(self, ctx, input):
         
         # Retrieve plant genome
         plant_genome = self.dfu.get_objects({'object_refs': [input['input_ws']+'/'+input['input_genome']]})['data'][0]
-        output = {'ftrs': len(plant_genome['data']['features'])}
+
+        # Force upgrade
+        if("feature_counts" in plant_genome['data']):
+            del(plant_genome['data']['feature_counts'])
+
+        USE_CDS=1
```
Comment (JamesJeffryes) [inline lib/kb_plant_rast/kb_plant_rastImpl.py]: Only use ALL_CAPS for globals set once not flags

**Review by JamesJeffryes — CHANGES_REQUESTED** [review]

**Pull request merged**